lib/hocon/impl/config_node_complex_value.rb

```diff
@@ -0,0 +1,53 @@
+# encoding: utf-8
+
+require 'hocon/impl'
+require 'hocon/impl/abstract_config_node_value'
+require 'hocon/impl/config_node_field'
+require 'hocon/impl/config_node_include'
+require 'hocon/impl/config_node_single_token'
+require 'hocon/impl/tokens'
+require 'hocon/config_error'
+
+class Hocon::Impl::ConfigNodeComplexValue < Hocon::Impl::AbstractConfigNodeValue
+  def initialize(children)
+    @children = children
+  end
+
+  attr_reader :children
+
+  def tokens
+    tokens = []
+    @children.each do |child|
+      tokens += child.tokens
+    end
+    tokens
```

Comment:
Not really important, but this might look cleaner as `token = @children.map { |child| child.tokens }`